### ISSUE TITLE
For some datasets, percent_better_than_baseline_all_objectives is huge because baseline values are infinitesimal

### ISSUE THREAD

**Issue by rpeck** [alteryx/evalml#1458]
I tested the `percent_better_than_baseline_all_objectives` values on a few datasets and found that `ExpVariance` on two of them was enormous. On `retail_data.csv` it's `1.4857769515459812e+18`.

Turns out this is because the cv objective scores for the baseline was infinitesimal:

```python
for scores in automl.results['pipeline_results'][0]['cv_data']:
    print(scores['all_objective_scores']['ExpVariance'])
```

```
0.0
0.0
1.1102230246251565e-16
```

With `evalml 0.13.2` these were all zero.

Now, obviously there's variability in this kind of thing, but it seems like `1.1102230246251565e-16` is a rounding error, not a real, valid value for this metric.

Here's the reproducer:

```python
import evalml
from evalml import AutoMLSearch
import pandas as pd

data = pd.read_csv('datasets/retail/retail_data.csv')
data = data.drop(['cancelled'], axis=1)

y = data['total']
X = data.drop(['total'], axis=1)

automl = AutoMLSearch(problem_type="regression", objective="Root Mean Squared Error", max_iterations=3)
automl.search(X, y)

# oh, I miss Scala
if automl.data_check_results is not None:
    for boom in automl.data_check_results:
        print(boom)

print(f'pipline 2 ExpVariance: {automl.results["pipeline_results"][2]["percent_better_than_baseline_all_objectives"]["ExpVariance"]}')


for scores in automl.results['pipeline_results'][0]['cv_data']:
    print(f'cv ExpVariance: {scores["all_objective_scores"]["ExpVariance"]}')

```

**Comment by dsherry** [comment]
Discussed with @rpeck . Next steps:
* I'll get a PR up to hard-code the explained variance to 0 for the baseline pipelines
* Doing so will mean that the pct improvement for exp variance will always be `nan` for the other pipelines
* We may wanna file a bug in sklearn about [exp var](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.explained_variance_score.html) misbehaving on this data / with this model

**Comment by dsherry** [comment]
Oops nope, just discussed with @freddyaboulton and he had a great alternative suggestion for point 1 above: update the pct improvement calculation to use `np.isclose`. Working on that now.
